### ISSUE TITLE
[Enhancement] Dashboards build pipeline on jenkins does not have snapshots builds

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/opensearch-build#1577]
Coming from #1497 we need dashboards to have snapshots builds before we can push it from ci to artifacts snapshots dir.
@kavilla will take a look on this per ticket.
Please refer to https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/distribution-build.jenkinsfile to add the snapshots section in https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch-dashboards/distribution-build.jenkinsfile

Thanks.

**Comment by bbarani** [comment]
@joshuarrrr @kavilla @peterzhuamazon Are we planning to add snapshot builds for OpenSearch Dashboards?

**Comment by gaiksaya** [comment]
Hi @kavilla Are you working on this issue?

**Comment by cwperks** [comment]
This would really help security-dashboards-plugin CI checks to run more efficiently. The checks are currently running into an issue where the runners are running out of disk space: https://github.com/opensearch-project/security-dashboards-plugin/issues/1611

More details on the CI checks can also be found here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5335

In order to run integ tests, security dashboards plugin checks out the OSD repo and runs `yarn osd bootstrap` to install the node modules and those steps collectively take multiple GBs of disk space.

**Comment by gaiksaya** [comment]
I believe we need the OpenSearch-Dashboards core repository to tell us how to build the snapshots version of core engine. For OpenSearch, we have https://github.com/opensearch-project/opensearch-build/blob/main/jenkins/opensearch/publish-min-snapshots.jenkinsfile that published opensearch-min-snapshots to artifacts.opensearch.org.
Something similar is what is expected I believe.  @cwperks How about create an issue in https://github.com/opensearch-project/OpenSearch-Dashboards to add the same and then next step would be integrating it with build pipelines?

**Comment by cwperks** [comment]
@gaiksaya I opened another issue in the OSD repo [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5336) about publishing snapshots to artifacts.opensearch.org.
